src/path.cr

```diff
@@ -0,0 +1,941 @@
+# A `Path` represents a filesystem path and allows path-handling operations
+# such as querying it's components as well as semantic manipulations.
+#
+# A path is hierarchical and composed of a sequence of directory and file name
+# elements separated by a special separator or delimiter. A root component,
+# that identifies a file system hierarchy, may also be present.
+# The name element that is farthest from the root of the directory hierarchy is
+# the name of a file or directory. The other name elements are directory names.
+# A Path can represent a root, a root and a sequence of names, or simply one or
+# more name elements.
+# A Path is considered to be an empty path if it consists solely of one name
+# element that is empty. Accessing a file using an empty path is equivalent
+# to accessing the default directory of the process.
+#
+# # Examples
```

Comment:
It would make sense to have more platform agnostic examples first since that should be the most used ones.

Reply:
Yes, I intend to update the examples and use more agnostic examples. See https://github.com/crystal-lang/crystal/pull/5635#issuecomment-420594345